rskj-core/src/main/java/org/ethereum/listener/GasPriceTracker.java

```diff
@@ -54,26 +53,46 @@ public class GasPriceTracker extends EthereumListenerAdapter {
     private static final BigInteger BI_10 = BigInteger.valueOf(10);
     private static final BigInteger BI_11 = BigInteger.valueOf(11);
 
-    private final Coin[] txWindow = new Coin[TX_WINDOW_SIZE];
-
     private final Double[] blockWindow = new Double[BLOCK_WINDOW_SIZE];
 
     private final AtomicReference<Coin> bestBlockPriceRef = new AtomicReference<>();
     private final BlockStore blockStore;
 
     private Coin defaultPrice = Coin.valueOf(20_000_000_000L);
-    private int txIdx = TX_WINDOW_SIZE - 1;
-
     private int blockIdx = 0;
 
-    private Coin lastVal;
+    private final GasCalculator gasCalculator;
 
-    private GasPriceTracker(BlockStore blockStore) {
+    private GasPriceTracker(BlockStore blockStore, GasCalculator gasCalculator) {
         this.blockStore = blockStore;
+        this.gasCalculator = gasCalculator;
     }
 
+    public static GasPriceTracker create(BlockStore blockStore, GasCalculator.GasCalculatorType gasCalculatorType) {
+        GasCalculator gasCal;
+        switch (gasCalculatorType) {
+            case WEIGHTED:
+                gasCal = new GasWeightedCalc();
+                break;
+            case LEGACY:
+                gasCal = new LegacyGasCalculator();
+                break;
+            default:
+                throw new IllegalArgumentException("Unknown gas calculator type: " + gasCalculatorType);
+        }
+        GasPriceTracker gasPriceTracker = new GasPriceTracker(blockStore, gasCal);
+        gasPriceTracker.initializeWindowsFromDB();
+
+        return gasPriceTracker;
+    }
+
+    /**
+     * @deprecated Use {@link #create(BlockStore, GasCalculator.GasCalculatorType)} instead.
+     */
+    @Deprecated
     public static GasPriceTracker create(BlockStore blockStore) {
-        GasPriceTracker gasPriceTracker = new GasPriceTracker(blockStore);
+        //Will be using the legacy gas calculator as default option
+        GasPriceTracker gasPriceTracker = new GasPriceTracker(blockStore, new LegacyGasCalculator());
```

Comment:
Just a small suggestion, I think here you can just call the new create method and pass `GasCalculator.GasCalculatorType.LEGACY` as the `gasCalculatorType` argument, that way you can just do `return create(blockStore, GasCalculator.GasCalculatorType.LEGACY)` and avoid repeating the following 2 lines.